src/trezor/utils.py

```diff
@@ -104,28 +104,20 @@ def obj_eq(l, r):
     """
     if l.__class__ is not r.__class__:
         return False
-    if hasattr(l, "__slots__"):
-        return obj_slots_dict(l) == obj_slots_dict(r)
-    else:
+    if not hasattr(l, "__slots__"):
         return l.__dict__ == r.__dict__
+    for slot in l.__slots__:
+        if getattr(l, slot, None) != getattr(r, slot, None):
+            return False
```

Comment:
This doesn't check that `r` has some slots that `l` does not have.

Reply:
technically `l.__slots__ is r.__slots__`, no?

Reply:
Yep! 👍

Reply:
> technically `l.__slots__ is r.__slots__`, no?

fixed, thanks!